src/lib/snark_worker/rpc_get_work.ml

```diff
@@ -12,10 +12,10 @@ module Master = struct
   let name = "get_work"
 
   module T = struct
-    type query = unit
+    type query = [ `V2 | `V3 ]
```

Comment:
Is the plan to support V2 and the older unpartitioned work specs long-term, or will we eventually drop those and switch to V3 and the partitioned specs completely?

Reply:
I hope we could drop old work spec. The ugliness of this PR train is largely due to the fact we want things to be compatible. You can see Work Selector is still exposed beyond via Work Partitioner, in particular via GraphQL queries.

This is expected in the first place as @georgeee requested there being no breaking change. 

In my ideal world Work Selector shouldn't be exposed in anyway expect the Partitioner Layer. But sadly it's not true now.